### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by initseis**
In this branch @bimbolabuari and I:

- Stored the people data in JSON files
- Stored the books data in JSON files
- Stored the rentals data in JSON files
- Load data from files on start up
- Handle the case when one or more files are missing

### CODE REVIEW

File: list_people.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
 class ListPeople
   def list_people(people)
     people.each do |p|
-      puts "[#{p.class}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+      if p['specialization']
+        puts "[teacher] Name: #{p['name']}, ID: #{p['id']}, Age: #{p['age']}"
+      else
+        puts "[student] Name: #{p['name']}, ID: #{p['id']}, Age: #{p['age']}"
+      end
```
Comment (DeliceLydia) [inline list_people.rb]: - [ ] Inside your `json` file I suggest that you add `json.class` like if is `people.json` file you can outputs:
`{"json_class":"Student","name":"Sam","age":"14","parent_permission":true}`
`{"json_class":"Teacher","name":"Fox","age":"56","parent_permission":true,"specialization":"Starships"}`

File: list_books.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class ListBooks
   def list_books(books)
     books.each do |b|
-      puts "Title: #{b.title}, Author: #{b.author}"
+      puts "Title: #{b['title']}, Author: #{b['author']}"
```
Comment (DeliceLydia) [inline list_books.rb]: - [ ] Please add a `json_class`

File: create_rental.rb [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
     puts ''
     puts 'Select a person from the following list by number (not id)'
     people.each.with_index do |p, index|
-      if p.instance_of?(Student)
-        puts "#{index}) [Student] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+      if p['specialization']
+        puts "#{index}) [Teacher] Name: #{p['name']}, ID: #{p['id']}, Age: #{p['age']}"
       else
-        puts "#{index}) [Teacher] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+        puts "#{index}) [Student] Name: #{p['name']}, ID: #{p['id']}, Age: #{p['age']}"
```
Comment (DeliceLydia) [inline create_rental.rb]: - [ ] Kindly add `json_class`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by oxenprogrammer — APPROVED** [review]
```markdown
Hi @sevinchek  and @bimbolabuari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**